subsys/bluetooth/host/att.c

```diff
@@ -2918,6 +2924,33 @@ static struct bt_att_chan *att_chan_new(struct bt_att *att, atomic_val_t flags)
 	return chan;
 }
 
+#if defined(CONFIG_BT_EATT)
+#if defined(CONFIG_BT_TESTING)
+uint16_t bt_att_num_eatt(struct bt_conn *conn)
+{
+	struct bt_att *att = att_get(conn);
+	struct bt_att_chan *chan;
+	uint16_t num_enhanced = 0;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&att->chans, chan, node) {
+		if (atomic_test_bit(chan->flags, ATT_ENHANCED)) {
+			num_enhanced++;
+		}
+	}
+
+	return num_enhanced;
+}
+#endif /* CONFIG_BT_TESTING */
+
+static void att_enhanced_connection_work_handler(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct bt_att *att = CONTAINER_OF(dwork, struct bt_att, connection_work);
+
+	bt_eatt_connect(att->conn, att->eatt_chans_to_connect);
```

Comment:
Should we check the return code?

Reply:
I am not sure if there is much to do here if this fails other than logging an error. If we are out of memory, channels should already be connected and they are unlikely to be freed before the connection terminates.

Reply:
We should log an error then :) 

If/when we have a EATT connect callback, that could also be called with error here.